llvm-general/src/LLVM/General/OrcJIT.hs

```diff
@@ -12,7 +12,12 @@ module LLVM.General.OrcJIT (
     mangleSymbol,
     withIRCompileLayer,
     withModuleSet,
-    withObjectLinkingLayer
+    withObjectLinkingLayer,
+    addModuleSet,
+    createObjectLinkingLayer,
```

Comment:
One of the goals of llvm-general is to handle C++ lifetime issues, so I've avoided exposing individual create/dispose functions but instead defined lots of withFoo functions. Unless it's provably necessary, do not expose the create/dispose functions.